### PR TITLE
Feature/build on launch

### PR DESCRIPTION
**Pull request by KoeMai**
### This changes behavior of launch and debug steps

The following changes are proposed:

- Add new configuration option `buildBeforeRun`

## The purpose of this change

This modification allow to configure the launchTargetPath, lanuchTarget or debugTarget to
build the target before execution.

## Open steps

- [x] Add option to documentation

### CODE REVIEW

File: src/cmake-tools.ts [hunk 1]
```diff
@@ -758,7 +758,43 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
       log.warning('No program will be executed');
       return null;
     }
-    return chosen.path;
+    return executable.path;
+  }
+
+  async PrepareLaunchTargetExecutable(): Promise<api.ExecutableTarget|null> {
```
Comment (vector-of-bool) [inline src/cmake-tools.ts]: Lowercase `prepare`

File: src/cmake-tools.ts [hunk 2]
```diff
@@ -793,19 +829,9 @@ export class CMakeTools implements vscode.Disposable, api.CMakeToolsAPI {
...
-        log.debug('Configuration of project failed.');
-        return null;
-      }
-    }
-    const target = await this.getOrSelectLaunchTarget();
-    if (!target) {
-      // The user has nothing selected and cancelled the prompt to select a target.
-      log.debug('No target selected.');
+
+    const targetExecutable = await this.PrepareLaunchTargetExecutable();
```
Comment (vector-of-bool) [inline src/cmake-tools.ts]: Compile error because this var was renamed.

File: src/cmake-tools.ts (hunk 1 above)
Comment (KoeMai) [reply src/cmake-tools.ts]: Done

File: src/cmake-tools.ts (hunk 2 above)
Comment (KoeMai) [reply src/cmake-tools.ts]: Done

**Pull request merged**